### PR TITLE
Cleanup some intern code.

### PR DESCRIPTION
**Pull request by postwait**
This should yield no functional change.  It adds some asserts and
macros for commonly repeated ops and reworks a few predicated to
avoid unsigned rolling (that should never have been possible anyway).
Make the test a bit more rigorous.

### CODE REVIEW

File: test/intern_test.c [hunk 1]
```diff
@@ -83,7 +101,13 @@ void load_words(void) {
   while(fgets(buff, sizeof(buff), fp)) {
     if(strlen(buff) < 1) continue;
     if(buff[strlen(buff)-1] == '\n') buff[strlen(buff)-1] = '\0';
-    words[word_cnt++] = strdup(buff);
+    int upsize = mtev_rand() % 4096;
+    if(upsize < strlen(buff)) upsize = strlen(buff);
+    char *str = malloc(upsize+1);
```
Comment (cody271) [inline test/intern_test.c]: Where do we call `free(str)`?

File: src/utils/mtev_intern.c [hunk 2]
```diff
@@ -322,7 +325,8 @@ borrow_free_node(mtev_intern_pool_t *pool,
   /* If we can manage to borrow the requested len without needing
    * to move this node into another freeslot, we can just "borrow"
    * the bytes from be done. */
-  while((oldsize = ck_pr_load_64(&n->size)) - len >= l->lsize) {
+  while((oldsize = ck_pr_load_64(&n->size)) >= l->lsize + len) {
```
Comment (cody271) [inline src/utils/mtev_intern.c]: I don't quite follow the changes here and below.. Why is `size >= lsize + len` better than `size - len >= lsize` ?

File: test/intern_test.c (hunk 1 above)
Comment (postwait) [reply test/intern_test.c]: It is not freed, it's building a table that's used until exit.

File: src/utils/mtev_intern.c (hunk 2 above)
Comment (postwait) [reply src/utils/mtev_intern.c]: they are unsigned types.  So A - B >= C could have very bad behavior if  B > A.. (underflow).

By adding, you guarantee that you'd have to overflow to break (which is impracticle given that lengths of allocations here are stored in a unsigned:23

**Pull request merged**